qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -392,4 +394,32 @@ public static Path createTempDir(String prefix) throws IOException {
         return Files.createTempDirectory(getRootTempDir(), prefix, NEW_DIR_PERMS);
     }
 
+    @FunctionalInterface
+    public interface ThrowingConsumer<T> {
+        void accept(T t) throws Exception;
+    }
+
+    public void withCustomConfig(ThrowingConsumer<Path> action) throws Exception {
+        Path tempDir = Files.createTempDirectory(getRootTempDir(), "custom-config");
+        Path tempConf = tempDir.resolve("elasticsearch");
+        FileUtils.copyDirectory(installation.config, tempConf);
+
+        Platforms.onLinux(() -> sh.run("chown -R elasticsearch:elasticsearch " + tempDir));
+
+        if (distribution.isPackage()) {
+            Files.copy(installation.envFile, tempDir.resolve("elasticsearch.bk"));// backup
+            append(installation.envFile, "ES_PATH_CONF=" + tempConf + "\n");
+        } else {
+            sh.getEnv().put("ES_PATH_CONF", tempConf.toString());
+        }
+
+        action.accept(tempConf);
```

Comment:
The previous implementation of the `withCustomConfig` method in `PackageTests.java` used exception handling to ensure that the cleanup / restore steps where always carried out. I'm curious about why this method isn't doing that?

Reply:
I don't think we need this type of exception handling at all, since the tests are single threaded, and all test execution stops once a test failure is found. So there is no reason to restore the state, since the exception means the test will have failed, and in fact restoring the state could make diagnosing the failure more difficult.